src/ol/view.js

```diff
@@ -256,10 +256,17 @@ ol.View.prototype.getCenter = function() {
 
 
 /**
+ * @param {Array.<number>=} opt_hints Destination array.
  * @return {Array.<number>} Hint.
  */
-ol.View.prototype.getHints = function() {
-  return this.hints_.slice();
+ol.View.prototype.getHints = function(opt_hints) {
```

Comment:
Ok, I agree we can't reuse the same hints array everywhere.  Though I think it is a pity that view hints are specific to a frame instead of a view.

If I ever get around to changing how animation works (so view properties are properly animated and we could finally have a "side-by-side" example that look nice), then I think we won't need the extra code here.